lib/msf/core/post/file.rb

```diff
@@ -322,6 +325,55 @@ def exist?(path)
 
   alias exists? exist?
 
+
+  #
+  # See if +path+ on the remote system is accessible in the manner specified by +perm+ by the current user.
+  #
+  # @param path [String] Remote path to check
+  # @param perms [Array] Array of permissions to check for
+  #
+  # @return [Boolean] true if +path+ is accessible to the current user via any of the manners listed in +perms+, otherwise false.
+  #
+  def check_win_path_permissions(path, perms)
+    # Grab permissions of the path in question and extract relevant info using a regex.
+    # Also strip extra whitespace before comparison occurs.
+    permissions_file_raw = cmd_exec("icacls \"#{path}\"").to_s
+    permissions_file_array = permissions_file_raw.scan(/^.+? (.+?\\.+?):(\(.*\))/)
```

Comment:
Same comment as for groups, a structured parser might be more maintainable.

Reply:
With https://github.com/rapid7/metasploit-framework/pull/15987/commits/f3f907b3f630547b2d4aef4296545ee33e1a8fe1 I've tried to restructure this logic and break it down into more reusable functions. Also added in an example of the output.

Let me know if this works for you or if you think it needs further adjustment.